config.go

```diff
@@ -136,6 +137,10 @@ type Config struct {
 	// LogOutput is the writer where logs should be sent. If this is not
 	// set, logging will go to stderr by default.
 	LogOutput io.Writer
+	// Logger is a custom logger which you provide. If Logger is set, it will use this for
+	// the internal logger. If this is not set, it will fall back to the behavior for
+	// LogOutput is not set, logging will go to stderr by default.
+	Logger *log.Logger
```

Comment:
Can we add a line break between this and the field above? The description could also be cleaned up to be more clear about the LogOutput writer being used when Logger is nil. We should also add a similar note to the LogOutput fie.